ciao-cli/main.go

```diff
@@ -27,13 +27,66 @@ import (
 	"net/http"
 	"os"
 	"strconv"
-	"time"
+	"text/template"
 
 	"github.com/golang/glog"
-
-	"github.com/01org/ciao/payloads"
 )
 
+// Item serves to represent a group of related commands
+type command struct {
+	SubCommands map[string]Command
+}
+
+// This is not used but needed to comply with Command interface
+func (c *command) ParseArgs(args []string) []string {
+	return args
+}
+
+func (c *command) Run(args []string) error {
+	cmdName := args[0]
+	subCommandName := args[1]
+	subCommand := c.SubCommands[subCommandName]
+	if subCommand == nil {
+		c.Usage(cmdName)
+	}
+	args = subCommand.ParseArgs(args[2:])
+	return subCommand.Run(args)
+}
+
+// Usage prints the available commands in an item
+func (c *command) Usage(name ...string) {
+	fmt.Fprintf(os.Stderr, `ciao-cli: Command-line interface for the Cloud Integrated Advanced Orchestrator (CIAO)
+
+Usage:
+
+	ciao-cli [options] `+name[0]+` sub-command [flags]
+`)
+
+	var t = template.Must(template.New("commandTemplate").Parse(commandTemplate))
+	t.Execute(os.Stderr, c)
+
+	fmt.Fprintf(os.Stderr, `
+Use "ciao-cli `+name[0]+` sub-command -help" for more information about that item.
+`)
+	os.Exit(2)
+}
+
+// Command is the interface that all cli commands should implement
+type Command interface {
+	Usage(...string)
```

Comment:
@albertomurillo So we don't need to export this interface, it's going to be a ciao-cli internal one.
As a consequence, it should `usage()`, `parseArgs` and `run()`.